### DISCUSSION TITLE
OAuth authentication suddenly fails

### DISCUSSION THREAD

**Discussion by lsalim19** [pocketbase/pocketbase#5206] [Q&A] [answered]
I was working on other parts of my app and hadn't test the authentication part recently. Now when I tried to authenticate by calling the JS library: 

```JS
const authData = await pb
        .collection(Collections.Users)
        .authWithOAuth2({ provider: provider });
```
I got an error. The log is like this:

```
ERROR POST /api/collections/users/auth-with-oauth2
└─ The requested resource wasn't found.
   └─ sql: no rows in result set
```

This happened after I empty my database tables of data but retain the tables and the permissions. Have I found a bug or is this some error in my part?

I'm using pocketbase JS library version 0.21.3 and pocketbase version 0.22.17 

Thanks.

**Comment by ganigeorgiev** [comment, marked as answer]
> This happened after I empty my database tables of data but retain the tables and the permissions.

I don't understand what do you mean by this. If you have manually cleared the `users` table without deleting also the `_externalAuths` references then I guess it is possible the auth action to response with the above error, but in any case I don't think your issue is PocketBase related.

If you think otherwise, please provide a **minimal** reproducible repo or at least some steps to reproduce and I'll try to have a look at it.

**Reply by lsalim19** [reply]
Yes, you are correct. I deleted the `_externalAuths` entries and now authentication works. Thank you.

**Reply by mrpn** [reply]
I had a similar issue – I changed the IDs in my users table using plain SQL, which caused me to lose the link to all existing oAuth references in the _externalAuths table.

To solve the issue, I simply used `$app.dao().db().newQuery("DELETE FROM _externalAuths").execute()`, and then everything worked again.
